### ISSUE TITLE
`tooltip` context not provided to screen reader users

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#4864]
### Summary

When accessed via keyboard the `tooltip` component's content is displayed visually, but not read back to screen reader users. As a result they don't receive feedback the `tooltip` has been accessed and/or opened.

### Actual Behavior

The `tooltip` content is not accessible or provided to screen reader users.

### Expected Behavior

The `tooltip` content should be read back politely, perhaps using the `aria-role='status'`, to provide context to screen reader users. 

Similar to the current functionality, after completion, the focus should return to the reference element.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/tooltip/#sample

### Reproduction Steps

1. Enable a screen reader (NVDA, JAWS or VoiceOver)
2. Navigate to the [sample](https://developers.arcgis.com/calcite-design-system/components/tooltip/#sample)
3. With the screen reader active, navigate via keyboard to focus on the "Hover for Tooltip" link to trigger the tooltip
4. Observe no context is provided to the screen reader, even if using the down arrow key to read back information

### Reproduction Version

beta.83

### Working W3C Example/Tutorial

https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html

### Relevant Info

- Windows (JAWS and NVDA)
- Mac OS (Safari/VoiceOver)
- related to: #4865

### Potential fix?
The `tooltip` content could be provided with a [`role='status'`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role) (giving it an `aria-live='polite'`).

**Comment by driskull** [comment]
@geospatialem the tooltip already has an aria-role of "tooltip". Do you have any examples/samples you can point to where it should be structured differently?

**Comment by geospatialem** [comment]
> @geospatialem the tooltip already has an aria-role of "tooltip". Do you have any examples/samples you can point to where it should be structured differently?
> 
> An example using tooltip role and status role would be helpful.

Great question! 

We'd want to add a few aria labels to the `Host` including:
- `aria-describedby={effectiveReferenceElement}` and 
- `aria-live="polite"` (this is the equivalent of `aria-role="polite"`)

In addition to removing the `aria-label` value so screen reader users don't hear duplicative information. As a result, this could make the `referenceElement` prop required and no longer require the `label` prop.

In order for screen readers to hear the content, content encompassed within the `calcite-tooltip` will need an HTML element with content, such as a `<p>`, `<ul>`, etc. otherwise if content is added it won't be read back to the screen reader. 

@driskull, what are your thoughts on if we could and should add in a `<p>` element (or similar) to the component? We could also modify our [docs sample](https://developers.arcgis.com/calcite-design-system/components/tooltip/#sample) to include a `<p>` tag indicating use, if that makes more sense, too.

Some samples of `tooltip` if the above approaches are implemented:

```html
<!-- Tooltip is read back to the screen reader with HTML element -->
<calcite-link id="tooltip-sr">Tooltip read to screen reader</calcite-link>
    <calcite-tooltip reference-element="tooltip-sr">
        <p>Tooltip content</p>
</calcite-tooltip>

<!-- Tooltip doesn't get read back to the screen reader (no HTML element resides in the tooltip) -->
<calcite-link id="tooltip-no-sr">Tooltip not read to screen reader</calcite-link>
    <calcite-tooltip reference-element="tooltip-no-sr">
        <p>Tooltip content</p>
</calcite-tooltip>

**Comment by driskull** [comment]
@geospatialem according to the tooltip role, the aria-describedby should go on the reference element and not the tooltip.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role

```html
<label for="password">Password:</label>
<input aria-describedby="passwordrules" id="password" type="password" />
<div role="tooltip" id="passwordrules">
  <p>
      Password Rules:
  </p>
  <ul>
    <li> Minimum of 8 characters</li>
    <li> Include at least one lowercase letter, one uppercase letter, one number and one special character</li>
    <li>Unique to this website</li>
  </ul>
</div>
```

> In order for the JAWS and NVDA screen readers to hear the content, content encompassed within the calcite-tooltip will need an HTML element with content, such as a <p>, <ul>, etc. otherwise if content is added it won't be read back to the screen reader (not a requirement of VoiceOver).

Could I add some kind of role instead of changing it to <p>? If users already slot a <p> I don't want to break their existing semantics.

> aria-live="polite"

I will add this.

**Comment by geospatialem** [comment]
> Could I add some kind of role instead of changing it to `<p>`? If users already slot a `<p>` I don't want to break their existing semantics.

I don't believe there is a role we could use, the way I understand it is if there is content present, the screen reader will continue reading it off. Otherwise it doesn't seem to be recognized. FWIW the only browser this seems to effect is Chrome, so we could update the docs site to include a `<p>` in the sample.

**Comment by benelan** [comment]
verified on master
